instrumentation/mysql2/lib/opentelemetry/instrumentation/mysql2/instrumentation.rb

```diff
@@ -30,6 +30,7 @@ class Instrumentation < OpenTelemetry::Instrumentation::Base
         option :peer_service, default: nil, validate: :string
         option :enable_sql_obfuscation, default: false, validate: :boolean
         option :db_statement, default: :include, validate: %I[omit include obfuscate]
+        option :span_name, default: :statement_type, validate: %I[statement_type db_name db_operation_and_name]
```

Comment:
Something feels weird to me about the `db_operation_and_name` option. I wonder if the option should communicate that it's reading off the `Mysql2.attributes` hash, somehow?  And writing that, maybe the option should _just_ defer to a `Mysql2.attributes` attribute and not include the `db_name`? And if users want to also inject db name, there could be another option for that? `%I[statement db_name mysql_attr enriched_mysql_attr]` or something. To be clear, I'm just musing, and don't think this is a blocker.

Reply:
The argument for `"#{db_operation} #{db_name}"` is [that that's what the spec suggests](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md):
> If `db.sql.table` is not available due to its semantics, the span SHOULD be named `<db.operation> <db.name>`.

Of course, that raises the question: can we get the table somehow without parsing?
> SQL spans SHOULD be named the following way, unless the statement is known to be of low cardinality: `<db.operation> <db.name>.<db.sql.table>`, provided that `db.operation` and `db.sql.table` are available.

Reply:
Ah, RTFS 😄 - makes sense.